service/history/transfer_queue_active_task_executor.go

```diff
@@ -896,6 +900,7 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 		inheritedBuildId,
 		initiatedEvent.GetUserMetadata(),
 		mutableState.GetExecutionInfo().GetVersioningInfo().GetVersioningOverride(),
+		mergePriorities(mutableState.GetExecutionInfo().Priority, attributes.Priority),
```

Comment:
I'm not sure about this one, I didn't see any mention of this case in the new API docs.

It seems reasonable to me for the child to inherit the parent's priority, but still be able to override it.